.github/workflows/tests.yaml

```diff
@@ -21,15 +21,15 @@ jobs:
       uses: actions/checkout@v3
 
     - name: Install Rust
-      uses: dtolnay/rust-toolchain@stable
-      with: 
-        toolchain: stable
+      uses: actions-rust-lang/setup-rust-toolchain@v1
+      with:
+        toolchain: nightly-2023-04-17
```

Comment:
If we are switching to this rust toolchain action can you update the other workflows as well. In addition, this action has `Swatinem/rust-cache` built in, can you update so we don't have redundant cache.